Sources/Smithy/Document.swift

```diff
@@ -95,33 +95,48 @@ public extension Document {
 extension Document {
 
     private var jsonObject: Any {
-        switch self {
-        case .array(let array):
-            return array.map { $0.jsonObject }
-        case .boolean(let bool):
-            return bool
-        case .number(let double):
-            return double
-        case .object(let object):
-            return object.mapValues { $0.jsonObject }
-        case .string(let string):
-            return string
-        case .null:
+        do {
+            switch self {
+            case .list:
+                return try asList().map { $0.jsonObject }
+            case .boolean:
+                return try asBoolean()
+            case .number:
+                return try asDouble()
+            case .map:
+                return try asStringMap().mapValues { $0.jsonObject }
+            case .string:
+                return try asString()
+            case .blob:
+                return try asBlob().base64EncodedString()
+            case .timestamp:
+                let formatter = DateFormatter()
```

Comment:
set the formatter's locale to `en_US_POSIX` so that the system's locale settings don't affect the timestamp that you render